core/commands/name/name.go

```diff
@@ -59,8 +70,76 @@ Resolve the value of a dnslink:
 	},
 
 	Subcommands: map[string]*cmds.Command{
-		"publish": PublishCmd,
-		"resolve": IpnsCmd,
-		"pubsub":  IpnsPubsubCmd,
+		"publish":       PublishCmd,
+		"resolve":       IpnsCmd,
+		"pubsub":        IpnsPubsubCmd,
+		"verify-record": IpnsVerifyRecordCmd,
+	},
+}
+
+var IpnsVerifyRecordCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Verifies an IPNS Record.",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("key", true, false, "The IPNS key to validate."),
+		cmds.FileArg("record", true, false, "The path to a file to be added to IPFS.").EnableStdin(),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		key := strings.TrimPrefix(req.Arguments[0], "/ipns/")
+
+		file, err := cmdenv.GetFileArg(req.Files.Entries())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		var b bytes.Buffer
+
+		_, err = io.Copy(&b, file)
+		if err != nil {
+			return err
+		}
+
+		var entry ipns_pb.IpnsEntry
+		err = proto.Unmarshal(b.Bytes(), &entry)
+		if err != nil {
+			return err
+		}
+
+		id, err := peer.Decode(key)
+		if err != nil {
+			return err
+		}
+
+		pub, err := id.ExtractPublicKey()
+		if err != nil {
+			return err
+		}
+
+		err = ipns.Validate(pub, &entry)
+		if err != nil {
+			return err
+		}
+
+		return cmds.EmitOnce(res, &entry)
+	},
+	Type: &ipns_pb.IpnsEntry{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *ipns_pb.IpnsEntry) error {
+			fmt.Fprintf(w, "Record is valid:\n\n")
+			fmt.Fprintf(w, "Value:\t\t%s\n", string(out.Value))
```

Comment:
nit: always use `%q`  when outputing untrusted values that came from data published by random people :)
```suggestion
			fmt.Fprintf(w, "Value:\t\t%q\n", string(out.Value))
```